include/fiction/technology/is_sidb_gate_design_impossible.hpp

```diff
@@ -0,0 +1,108 @@
+//
+// Created by Jan Drewniok on 25.10.23.
+//
+
+#ifndef FICTION_IS_SIDB_GATE_DESIGN_IMPOSSIBLE_HPP
+#define FICTION_IS_SIDB_GATE_DESIGN_IMPOSSIBLE_HPP
+
+#include "fiction/algorithms/iter/bdl_input_iterator.hpp"
+#include "fiction/algorithms/simulation/sidb/is_operational.hpp"
+#include "fiction/algorithms/simulation/sidb/sidb_simulation_engine.hpp"
+#include "fiction/algorithms/simulation/sidb/sidb_simulation_parameters.hpp"
+#include "fiction/technology/cell_technologies.hpp"
+#include "fiction/technology/charge_distribution_surface.hpp"
+#include "fiction/technology/physical_constants.hpp"
+#include "fiction/technology/sidb_charge_state.hpp"
+#include "fiction/technology/sidb_defects.hpp"
+#include "fiction/traits.hpp"
+
+#include <cassert>
+#include <vector>
+
+namespace fiction
+{
+/**
+ * This struct contains parameters to determine if SiDB gate design is impossible.
+ */
+struct is_gate_design_impossible_params
+{
+    /**
+     * All parameters for physical SiDB simulations.
+     */
+    sidb_simulation_parameters phys_params{};
+    /**
+     * The simulation engine to be used for simulation.
+     */
+    sidb_simulation_engine sim_engine{sidb_simulation_engine::QUICKEXACT};
+};
+/**
+ * This function evaluates whether it is impossible to design a SiDB gate for a given truth table in the given layout
+ * due to atomic defects. It determines the possible charge states at the output BDL pairs. Atomic defects can cause a
+ * BDL pair to be neutrally charged only. Thus, the BDL pair would not work as intended.
+ *
```

Comment:
I'm not sure I understand how this function is working from the documentation alone. You are already passing a layout. So is that just the surface? In that case, maybe use `sidb_defect_surface` (I hope that's the currently correct name). Or are you passing a skeleton? If so, it would be nice to mention that.